cpp/src/parser/expression.cpp

```diff
@@ -0,0 +1,389 @@
+#include <charconv>
+#include <sstream>
+
+#include "lexer.hh"
+#include "expression.hpp"
+#include "statement.hpp"
+
+// Expression
+
+ExpressionP Expression::parse(Lexer& lexer)
+{
+	auto tokenType = lexer.type();
+	switch (tokenType) {
+		case TokenType::If:
+			return IfStatement::parse(lexer);
+	}
+
+	return parseEquality(lexer);
+}
+
+ExpressionP Expression::parseEquality(Lexer& lexer)
+{
+	auto left = parseCompare(lexer);
+
+	while (true) {
+		auto tokenType = lexer.type();
+		switch (tokenType) {
+			case TokenType::Eq:
```

Comment:
Since we're using C++20, can we add the `[[fallthrough]];` attribute to the switch cases where it applies?